### ISSUE TITLE
[bug] conan install serf fails in ConanFile.run()

### ISSUE THREAD

**Issue by malchera** [conan-io/conan#10531]
When attempting to install the serf library e.g. with
`conan install serf/1.3.9@ --build=serf`
The build fails with the following error message:
```
[...]
serf/1.3.9: Generator txt created conanbuildinfo.txt
serf/1.3.9: Aggregating env generators
serf/1.3.9: Calling build()
Warning:  ignoring -m option
usage: scons [OPTION] [TARGET] ...

SCons Error: no such option: -6
serf/1.3.9:
serf/1.3.9: ERROR: Package 'de0d7a09fb3ef801ded4c3cc223c5928a5f42e36' build failed
serf/1.3.9: WARN: Build folder /home/myuser/.conan/data/serf/1.3.9/_/_/build/de0d7a09fb3ef801ded4c3cc223c5928a5f42e36
ERROR: serf/1.3.9: Error in build() method, line 111
        self.run("scons {} {}".format(" ".join(escape_str(s) for s in args), " ".join("{}={}".format(k, escape_str(v)) for k, v in kwargs.items())), run_environment=True)
        ConanException: Error 2 while executing scons "-Y" "/home/myuser/.conan/data/serf/1.3.9/_/_/build/de0d7a09fb3ef801ded4c3cc223c5928a5f42e36/source_subfolder" APR="/home/myuser/.conan/data/apr/1.7.0/_/_/package/435690f0682a7e600daf767c97acf6f625b77db0" APU="/home/myuser/.conan/data/apr-util/1.6.1/_/_/package/15462a466576cd30e6c375ccf92dc4b262749330" OPENSSL="/home/myuser/.conan/data/openssl/1.1.1g/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709" PREFIX="/home/myuser/.conan/data/serf/1.3.9/_/_/package/de0d7a09fb3ef801ded4c3cc223c5928a5f42e36" LIBDIR="/home/myuser/.conan/data/serf/1.3.9/_/_/package/de0d7a09fb3ef801ded4c3cc223c5928a5f42e36/lib" ZLIB="/home/myuser/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709" DEBUG="False" APR_STATIC="True" CFLAGS="-fPIC -m64 -O3 -s" LINKFLAGS=" -L'/home/myuser/.conan/data/apr-util/1.6.1/_/_/package/15462a466576cd30e6c375ccf92dc4b262749330/lib' -L'/home/myuser/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib' -L'/home/myuser/.conan/data/openssl/1.1.1g/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib' -L'/home/myuser/.conan/data/apr/1.7.0/_/_/package/435690f0682a7e600daf767c97acf6f625b77db0/lib' -L'/home/myuser/.conan/data/libiconv/1.16/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/lib' -L'/home/myuser/.conan/data/expat/2.4.1/_/_/package/c215f67ac7fc6a34d9d0fb90b0450016be569d86/lib'" CPPFLAGS="-DXML_STATIC -DAPR_DECLARE_STATIC -DAPU_DECLARE_STATIC -DNDEBUG -I'/home/myuser/.conan/data/apr-util/1.6.1/_/_/package/15462a466576cd30e6c375ccf92dc4b262749330/include' -I'/home/myuser/.conan/data/zlib/1.2.11/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include' -I'/home/myuser/.conan/data/openssl/1.1.1g/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include' -I'/home/myuser/.conan/data/apr/1.7.0/_/_/package/435690f0682a7e600daf767c97acf6f625b77db0/include' -I'/home/myuser/.conan/data/libiconv/1.16/_/_/package/6af9cc7cb931c5ad942174fd7838eb655717c709/include' -I'/home/myuser/.conan/data/expat/2.4.1/_/_/package/c215f67ac7fc6a34d9d0fb90b0450016be569d86/include'" CC="gcc" SOURCE_LAYOUT="False"
```

However if I c&p the problematic `scons` command into my bash terminal, it runs fine and builds libserf as expected.
A quick look into the `conanfile.py` shows that the line causing the issue is the following (inside the `build()` method):

```
with tools.chdir(self._source_subfolder):
    with self._build_context():
            self.run("scons {} {}".format(" ".join(escape_str(s) for s in args), " ".join("{}={}".format(k, escape_str(v)) for k, v in kwargs.items())), run_environment=True)
```
Apparently, somewhere downstream the parameters are spoiled, so that the command fails.
Btw, for the sake of testing I have printed the `scons` command that is passed to `self.run()` to the screen, c&p of this also works fine.
I'm not sure where the `-6` that `scons` complains about it coming from. This looks like a bug to me.

### Environment Details (include every applicable attribute)
  * Operating System+version: WSL2 + RHEL7.9
  * Compiler+version: gcc 9.3.0 (WSL2), 9.2.0 (RHEL7)
  * Conan version: 1.43.2 (WSL2), 1.45.0 (RHEL7)
  * Python version: 3.8.10 (WSL2), 3.9.0 (RHEL7)

### Steps to reproduce (Include if Applicable)
`conan install serf/1.3.9@ --build=serf`

**Comment by memsharded** [comment]
Quick try in Windows: everything seem to work fine here (there are some concerning warnings about .pdb files, when building in build_type=Release, but that is probably another issue.

Will try to test under linux now.

**Comment by memsharded** [comment]
reproduced in Linux, but also copying and pasting that line in the command line directly. Investigating a bit further, it seems the offending flag is ``CFLAGS="-fPIC -m64 -O3 -s"``. The ``-m64 is being the "6" error, and if you remove it, it will error on the "O" too. Something like ``scons "-Y" CFLAGS="-fPIC -m64 -O3 -s"`` fails with that error message.

I will check why ``AutotoolsBuildEnvironment`` is passing those, before moving this as a recipe issue to conan-center-index. But if @uilianries or @SSE4 could have a look to conan-center-index CI and try to understand why this recipe is correctly building there.

**Comment by uilianries** [comment]
Finally I found the problem, it's the scons package in Conan Center ...

When running with Conan, I can reproduce the same reported error:

```
$ /home/uilian/.conan/data/scons/4.2.0/_/_/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/bin/scons "-Y" "/home/uilian/.conan/data/serf/1.3.9/_/_/build/d93f51e80ee814b31d1274d08d479004c2edc126/source_subfolder" "-j4" CFLAGS="-fPIC -m64 -O3 -s" LINKFLAGS=" -L'/home/uilian/.conan/data/apr-util/1.6.1/_/_/package/4829707b0cffa0be033d0c425d4ff7445de4b922/lib' -L'/home/uilian/.conan/data/zlib/1.2.11/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/lib' -L'/home/uilian/.conan/data/openssl/1.1.1m/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/lib' -L'/home/uilian/.conan/data/apr/1.7.0/_/_/package/5a93bf7d483e99fe660ecfa6f981f28ff2854b44/lib' -L'/home/uilian/.conan/data/libiconv/1.16/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/lib' -L'/home/uilian/.conan/data/expat/2.4.1/_/_/package/2bf99ee01cda593d5986afc64acf262ad576354f/lib'" CPPFLAGS="-DXML_STATIC -DAPR_DECLARE_STATIC -DAPU_DECLARE_STATIC -DNDEBUG -I'/home/uilian/.conan/data/apr-util/1.6.1/_/_/package/4829707b0cffa0be033d0c425d4ff7445de4b922/include' -I'/home/uilian/.conan/data/zlib/1.2.11/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/include' -I'/home/uilian/.conan/data/openssl/1.1.1m/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/include' -I'/home/uilian/.conan/data/apr/1.7.0/_/_/package/5a93bf7d483e99fe660ecfa6f981f28ff2854b44/include' -I'/home/uilian/.conan/data/libiconv/1.16/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/include' -I'/home/uilian/.conan/data/expat/2.4.1/_/_/package/2bf99ee01cda593d5986afc64acf262ad576354f/include'" CC="gcc" APR="/home/uilian/.conan/data/apr/1.7.0/_/_/package/5a93bf7d483e99fe660ecfa6f981f28ff2854b44" APU="/home/uilian/.conan/data/apr-util/1.6.1/_/_/package/4829707b0cffa0be033d0c425d4ff7445de4b922" OPENSSL="/home/uilian/.conan/data/openssl/1.1.1m/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646" PREFIX="/home/uilian/.conan/data/serf/1.3.9/_/_/package/d93f51e80ee814b31d1274d08d479004c2edc126" LIBDIR="/home/uilian/.conan/data/serf/1.3.9/_/_/package/d93f51e80ee814b31d1274d08d479004c2edc126/lib" ZLIB="/home/uilian/.conan/data/zlib/1.2.11/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646" DEBUG="False" APR_STATIC="True" SOURCE_LAYOUT="False"
Warning:  ignoring -m option
usage: scons [OPTION] [TARGET] ...

SCons Error: no such option: -6
```

Now, if I ONLY change the scons binary, from my system:

```
$ pip show scons
Name: SCons
Version: 4.2.0
Summary: Open Source next-generation build tool.
Home-page: http://www.scons.org/
Author: William Deegan
Author-email: bill@baddogconsulting.com
License: MIT
Location: /opt/pyenv/versions/3.8.8/lib/python3.8/site-packages
Requires: setuptools
$ scons "-Y" "/home/uilian/.conan/data/serf/1.3.9/_/_/build/d93f51e80ee814b31d1274d08d479004c2edc126/source_subfolder" "-j4" CFLAGS="-fPIC -m64 -O3 -s" LINKFLAGS=" -L'/home/uilian/.conan/data/apr-util/1.6.1/_/_/package/4829707b0cffa0be033d0c425d4ff7445de4b922/lib' -L'/home/uilian/.conan/data/zlib/1.2.11/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/lib' -L'/home/uilian/.conan/data/openssl/1.1.1m/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/lib' -L'/home/uilian/.conan/data/apr/1.7.0/_/_/package/5a93bf7d483e99fe660ecfa6f981f28ff2854b44/lib' -L'/home/uilian/.conan/data/libiconv/1.16/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/lib' -L'/home/uilian/.conan/data/expat/2.4.1/_/_/package/2bf99ee01cda593d5986afc64acf262ad576354f/lib'" CPPFLAGS="-DXML_STATIC -DAPR_DECLARE_STATIC -DAPU_DECLARE_STATIC -DNDEBUG -I'/home/uilian/.conan/data/apr-util/1.6.1/_/_/package/4829707b0cffa0be033d0c425d4ff7445de4b922/include' -I'/home/uilian/.conan/data/zlib/1.2.11/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/include' -I'/home/uilian/.conan/data/openssl/1.1.1m/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/include' -I'/home/uilian/.conan/data/apr/1.7.0/_/_/package/5a93bf7d483e99fe660ecfa6f981f28ff2854b44/include' -I'/home/uilian/.conan/data/libiconv/1.16/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646/include' -I'/home/uilian/.conan/data/expat/2.4.1/_/_/package/2bf99ee01cda593d5986afc64acf262ad576354f/include'" CC="gcc" APR="/home/uilian/.conan/data/apr/1.7.0/_/_/package/5a93bf7d483e99fe660ecfa6f981f28ff2854b44" APU="/home/uilian/.conan/data/apr-util/1.6.1/_/_/package/4829707b0cffa0be033d0c425d4ff7445de4b922" OPENSSL="/home/uilian/.conan/data/openssl/1.1.1m/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646" PREFIX="/home/uilian/.conan/data/serf/1.3.9/_/_/package/d93f51e80ee814b31d1274d08d479004c2edc126" LIBDIR="/home/uilian/.conan/data/serf/1.3.9/_/_/package/d93f51e80ee814b31d1274d08d479004c2edc126/lib" ZLIB="/home/uilian/.conan/data/zlib/1.2.11/_/_/package/dfbe50feef7f3c6223a476cd5aeadb687084a646" DEBUG="False" APR_STATIC="True" SOURCE_LAYOUT="False"
scons: Reading SConscript files ...
Warning: Used unknown variables: SOURCE_LAYOUT
Checking for GNU-compatible C compiler...yes
scons: done reading SConscript files.
scons: Building targets ...
```

In summary, this issue should be moved to Conan Center Index.

Why did we see it only now? Because there are only 2 packages using SCons so far, and `serf` is the only one which passes options with white spaces. Anyway, those parameters are not honored, the SConstruct from serf overrides all CFLAGS.

**Comment by uilianries** [comment]
Opened the issue https://github.com/conan-io/conan-center-index/issues/9308 asking for scons fix
